### PR TITLE
Update readme and implement builder pattern

### PR DESCRIPTION
**Pull request by abhriyaroy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # RxLowpoly
```
Comment (aritraroy) [inline README.md:4]: The image subtitle can be "Fast, Beautiful and Customizable"

File: README.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
...
 
- An android library to convert your dull normal images into awesome ones with a crystallized lowpoly effect.
- <br>
+ An android library to convert your dull ordinary images into awesome
```
Comment (aritraroy) [inline README.md]: "Android" with A capital

Comment (aritraroy) [reply README.md]: An Android library to bring your ordinary photos to life with an awesome crystalize effect.

File: README.md [hunk 3]
```diff
@@ -2,8 +2,8 @@
...
+ ones with a crystallized lowpoly effect. <br>
```
Comment (aritraroy) [inline README.md:6]: Add badges for JCenter with library version. You can check examples from my libraries.

File: README.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
...
 
 ## Table of Contents
```
Comment (aritraroy) [inline README.md:7]: Installation should come above Library Details

File: README.md [hunk 5]
```diff
@@ -20,27 +20,36 @@
 ## Introduction
 
  LowpolyRxJava serves as an improvement over [XLowPoly](https://github.com/xyzxqs/XLowPoly) by 
-  - fixing `out of memory` crashes by scaling down the image losslessly before processing.
-  - providing better quality results by using `4000` as the point count by default which provides a good trade-off between speed and time.
-  - the higher point count leads to a longer execution period, but it is significantly reduced by `scaling down the image` before processing.
-  - provides wider choice of input sources like `bitmap`, `file`, `uri` or `drawable resource`.
-  - natively using `RxJava` for background processing thereby reducing boilerplate code on the developer's end.
+ -  fixing `out of memory` crashes by scaling down the image in a
```
Comment (aritraroy) [inline README.md]: Use bold instead of code for highlighting important words

Comment (aritraroy) [reply README.md]: USe code only when they refer directly as code

File: README.md [hunk 6]
```diff
@@ -20,27 +20,36 @@
...
+    loss-less manner before processing.
+ -  providing better quality results by using `4000` as the point count
+    by default which provides a good trade-off between speed and time.
+ -  the higher point count leads to a longer execution period, but it is
+    significantly reduced by `scaling down the image` before processing.
+ -  provides wider choice of input sources like `bitmap`, `file`, `uri`
```
Comment (aritraroy) [inline README.md]: "Bitmap", "File", etc. It is good to use code here as it directly refers to code. But "bitmap" should be "Bitmap" as the class name is that. Change the other ones also.

File: README.md [hunk 7]
```diff
@@ -20,27 +20,36 @@
...
+ <img src="https://i.imgur.com/iCddK29.jpg" alt="Original" width=400 height=250> | <img src="https://i.imgur.com/Z4zOgqH.jpg"  alt="Lowpoly" width=400 height=250>
  <img src="https://i.imgur.com/C5wzAqx.jpg" alt="Original" width=400 height=250> | <img src="https://i.imgur.com/mLjjrax.jpg"  alt="Lowpoly" width=400 height=250>
  <img src="https://i.imgur.com/Ho86fyo.jpg" alt="Original" width=400 height=250> | <img src="https://i.imgur.com/pm8MV8m.png"  alt="Lowpoly" width=400 height=250>
  <img src="https://i.imgur.com/D4DP8fu.jpg" alt="Original" width=400 height=250> | <img src="https://i.imgur.com/1zgjCyE.jpg"  alt="Lowpoly" width=400 height=250>
 							  
 ## Library Details
 
- - LowpolyRxJava uses [JNI](#jni) with 64 bit support to meet google specified requirement for all apps to be 64 bit enabled by August      2019.
+ - LowpolyRxJava uses [JNI](#jni) with 64 bit support to meet google
+   specified requirement for all apps to be 64 bit enabled by August
```
Comment (aritraroy) [inline README.md:67]: Link to google official website doc

File: README.md [hunk 8]
```diff
@@ -52,7 +61,12 @@
  
  ### Delaunay Triangulation
 
-  We take a set P of discrete points on an image plane P and apply <a href="https://en.wikipedia.org/wiki/Delaunay_triangulation">Delaunay Triangulation</a> DT(P) to produce triangles connecting 3 points   at a time such that no point in P is inside the circumcircle of any triangle in DT(P). These separate triangles taken together in-turn   provide us with the image having a crystallized effect.
+  We take a set P of `discrete points` on an image plane P and apply <a
```
Comment (aritraroy) [inline README.md]: Don't use code for discrete points. Is it code? Use bold or italic to highlight these things.

File: README.md (hunk 1 above)
Comment (aritraroy) [reply README.md:4]: Better to try with another version of the image with white background as not many people use darkreader.

File: README.md [hunk 9]
```diff
@@ -82,7 +96,7 @@
 
  That's it! <br>
```
Comment (aritraroy) [inline README.md]: "Step 1 -" and "Step 2 -" and not full stop after it

Comment (aritraroy) [reply README.md]: Do not use the version hardcoded here, com.github.abhriyaroy:LowpolyRxJava:1.0.1. Use badges for it. Check other libraries on how they do it, or check some of my libraries.

File: README.md [hunk 10]
```diff
@@ -82,7 +96,7 @@
...
 
-  Please note that using this library, it is assumed that RxJava and RxAndroid are already added as dependencies in your project but in-   case you don't have these dependencies, please add the following dependencies to your app module's build.gradle file :-
+  Please note that using this library, it is assumed that RxJava and RxAndroid are already added as dependencies in your project but in-case you don't have these dependencies, please add the following dependencies to your app module's build.gradle file :-
```
Comment (aritraroy) [inline README.md]: This should not be put separately by the users. These dependencies can be brought transitively from RxLowPoly itself.

File: README.md [hunk 11]
```diff
@@ -152,38 +165,48 @@ A `bitmap` of the generated lowpoly image is always returned irrespective of `sy
   
 ## Critical Analysis
 
-The following tests have been performed on a` Xiaomi Redmi Note 5 Pro with 6 gb Ram`. <br>
+The following tests have been performed on a `Xiaomi Redmi Note 5 Pro with 6 gb Ram`. <br>
 
  
-Input | Output | Input Source | Output Type | Quality | Time Required (in millis)
```
Comment (aritraroy) [inline README.md:158]: Output type seems redundant

Comment (aritraroy) [reply README.md:158]: The image thumbnails look squashed also. Removing one column can help here.

File: README.md [hunk 12]
```diff
@@ -152,38 +165,48 @@ A `bitmap` of the generated lowpoly image is always returned irrespective of `sy
   
 ## Critical Analysis
 
-The following tests have been performed on a` Xiaomi Redmi Note 5 Pro with 6 gb Ram`. <br>
+The following tests have been performed on a `Xiaomi Redmi Note 5 Pro with 6 gb Ram`. <br>
```
Comment (aritraroy) [inline README.md]: The `code` syntax is overused a lot. Use it only when it is code otherwise use bold or italic to highlight.

File: README.md [hunk 13]
```diff
@@ -152,38 +165,48 @@ A `bitmap` of the generated lowpoly image is always returned irrespective of `sy
...
+
+
+
+Thus it is evident that when quality is set to `High`, a good trade-off between speed and texture is obtained, hence the default value of `Quality` is set to `HIGH`.<br>
+Also, we can see that `bitmap` is the input format of choice as it is processed the fastest, followed by `Uri`, `File` and `Drawable` respectively.
 
 ## Sample App
 
-The <a href="https://play.google.com/store/apps/details?id=com.zebrostudio.lowpolyrx">sample app</a> provides an implementation of various configurations of `RxLowpoly` which one can experience first hand before using the library.
+The <a href="https://play.google.com/store/apps/details?id=com.zebrostudio.lowpolyrx">sample app</a> provides an implementation of various configurations of `RxLowpoly` which one can experience and assess first hand before using the library.
```
Comment (aritraroy) [inline README.md]: Link to the sample app activity link in the repo.

File: README.md [hunk 14]
```diff
@@ -152,38 +165,48 @@ A `bitmap` of the generated lowpoly image is always returned irrespective of `sy
...
+-  Screenshots :-
```
Comment (aritraroy) [inline README.md]: Use headline (H2 or H3) for screenshot and don't use ":-"

File: app/src/main/java/com/zebrostudio/rxlowpoly/examplefragments/BitmapAsyncFragment.kt [hunk 15]
```diff
@@ -70,7 +70,8 @@ class BitmapAsyncFragment : BaseFragment() {
   ): Single<Bitmap> {
     return RxLowpoly.with(activity!!.applicationContext)
       .input(bitmap)
-      .overrideScaling(downScalingFactor, maximumWidth)
+      .overrideScaling(downScalingFactor)
+      .overrideScaling(maximumWidth)
```
Comment (aritraroy) [inline app/src/main/java/com/zebrostudio/rxlowpoly/examplefragments/BitmapAsyncFragment.kt:74]: Why is it used twice?

File: app/src/main/java/com/zebrostudio/rxlowpoly/examplefragments/FileSyncFragment.kt [hunk 16]
```diff
@@ -99,7 +99,8 @@ class FileSyncFragment : BaseFragment() {
     return Single.create {
       val bitmap = RxLowpoly.with(activity!!.applicationContext)
         .input(inputFile)
-        .overrideScaling(downScalingFactor, maximumWidth)
+        .overrideScaling(downScalingFactor)
+      .overrideScaling(maximumWidth)
```
Comment (aritraroy) [inline app/src/main/java/com/zebrostudio/rxlowpoly/examplefragments/FileSyncFragment.kt:103]: Check all the formatting isssues

File: README.md [hunk 17]
```diff
@@ -19,40 +19,54 @@
 
 ## Introduction
```
Comment (aritraroy) [inline README.md:19]: The output directory is named as "Outputs" but should be named "output"

File: rxlowpoly/src/main/java/com/zebrostudio/rxlowpoly/internal/builder/RxLowpolyBuilder.kt [hunk 18]
```diff
@@ -99,25 +99,6 @@ class RxLowpolyBuilder {
     return this
   }
 
-  /**
```
Comment (aritraroy) [inline rxlowpoly/src/main/java/com/zebrostudio/rxlowpoly/internal/builder/RxLowpolyBuilder.kt:102]: The UriHelper class is outside the module. Should be inside one of the modules.

**Review by aritraroy — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (abhriyaroy) [reply README.md]: Updated summary

File: README.md (hunk 4 above)
Comment (abhriyaroy) [reply README.md:7]: Changed the order.

File: README.md (hunk 5 above)
Comment (abhriyaroy) [reply README.md]: Fixed.

File: README.md (hunk 7 above)
Comment (abhriyaroy) [reply README.md:67]: Added link.

**Pull request merged**